core/invoker/src/main/scala/whisk/core/invoker/InvokerReactive.scala

```diff
@@ -81,53 +75,14 @@ class InvokerReactive(config: WhiskConfig, instance: InstanceId, producer: Messa
     new MessageFeed("activation", logging, consumer, maximumContainers, 500.milliseconds, processActivationMessage)
   })
 
-  /** Initialize container clients */
-  implicit val docker = new DockerClientWithFileAccess()(ec)
-  implicit val runc = new RuncClient(ec)
-
-  /** Cleans up all running wsk_ containers */
-  def cleanup() = {
-    val cleaning = docker.ps(Seq("name" -> s"wsk${instance.toInt}_"))(TransactionId.invokerNanny).flatMap {
-      containers =>
-        val removals = containers.map { id =>
-          runc
-            .resume(id)(TransactionId.invokerNanny)
-            .recoverWith {
-              // Ignore resume failures and try to remove anyway
-              case _ => Future.successful(())
-            }
-            .flatMap { _ =>
-              docker.rm(id)(TransactionId.invokerNanny)
-            }
-        }
-        Future.sequence(removals)
-    }
-
-    Await.ready(cleaning, 30.seconds)
-  }
-  cleanup()
-  sys.addShutdownHook(cleanup())
-
   /** Factory used by the ContainerProxy to physically create a new container. */
   val containerFactory =
-    (tid: TransactionId, name: String, actionImage: ImageName, userProvidedImage: Boolean, memory: ByteSize) => {
-      val image = if (userProvidedImage) {
-        actionImage.publicImageName
-      } else {
-        actionImage.localImageName(config.dockerRegistry, config.dockerImagePrefix, Some(config.dockerImageTag))
-      }
-
-      DockerContainer.create(
-        tid,
-        image = image,
-        userProvidedImage = userProvidedImage,
-        memory = memory,
-        cpuShares = config.invokerCoreShare.toInt,
-        environment = Map("__OW_API_HOST" -> config.wskApiHost),
-        network = config.invokerContainerNetwork,
-        dnsServers = config.invokerContainerDns,
-        name = Some(name))
-    }
+    SpiLoader
+      .get[ContainerFactoryProvider]
+      .getContainerFactory(actorSystem, logging, config, instance)
+  val containerFactoryFunction = containerFactory.createContainer _
```

Comment:
No need to create an additional `val` I think, just put `containerFactory.createContainer _` into `props` directly.